### PR TITLE
Updated logic and unit test for ParseNormalizedSnapshotID function

### PR DESCRIPTION
**Pull request by jackieung-dell**
# Description
Updated logic and unit test for ParseNormalizedSnapshotID function to prevent a panic error message when accessing an index out of range.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

**Review by jooseppi-luna — APPROVED** [review]
lgtm, good work!

**Comment by gallacher** [comment]
I see some yaml linting failures that need to be addressed as well

**Comment by gallacher** [comment]
Coverage needs to improve for the following

coverage: 67.6% of statements
ok  	github.com/dell/csi-isilon/v2/service/interceptor	7.374s	coverage: 67.6% of statements

**Review by abhi16394 — COMMENTED** [review]
Please run cert-csi test and add the result since it has logic change

File: common/utils/utils_test.go [hunk 1]
```diff
@@ -81,6 +81,23 @@ func TestParseNormalizedVolumeID(t *testing.T) {
...
+	snapshotID, clusterName, accessZone, err := ParseNormalizedSnapshotID(ctx, "284=_=_=cluster1=_=_=System")
+
+	assert.Equal(t, "284", snapshotID)
+	assert.Equal(t, "cluster1", clusterName)
+	assert.Equal(t, "System", accessZone)
+	assert.Nil(t, err)
+
+	expectedError := "access zone not found in snapshot ID '284=_=_=cluster1'"
+	_, _, _, err = ParseNormalizedSnapshotID(ctx, "284=_=_=cluster1")
+	if err.Error() != expectedError {
```
Comment (santhoshatdell) [inline common/utils/utils_test.go:96]: Can you check that error is not nil and then check its message?

```
_, _, _, err = ParseNormalizedSnapshotID(ctx, "284=_=_=cluster1")
assert.NotNil(t, err)
assert.Contains(t, err.Error(), "access zone not found in snapshot ID")
```

**Review by bpjain2004 — DISMISSED** [review]

**Comment by santhoshatdell** [comment]
Converting to Draft since we are working on improving code coverage.

**Comment by santhoshatdell** [comment]
@jackieung-dell commits are not signed. Will re-open the PR with that addressed.

**Comment by santhoshatdell** [comment]
Closing this PR. Changes are ported to https://github.com/dell/csi-powerscale/pull/302

**Pull request closed without merging**